config/src/main/java/com/typesafe/config/impl/Parser.java

```diff
@@ -1063,6 +1063,7 @@ private static Path parsePathExpression(Iterator<Token> expression,
     private static Path parsePathExpression(Iterator<Token> expression,
             ConfigOrigin origin, String originalText) {
         // each builder in "buf" is an element in the path.
+        ArrayList<Token> pathTokens = new ArrayList<Token>();
```

Comment:
I think that's doable, but I think in that case, a `ConfigNodePath` is going to need to be passed into the `setValueOnPath()` method in `ConfigNodeComplexValue`, since we'll need the tokens to add in a new field.